kv.go

```diff
@@ -0,0 +1,142 @@
+package geche
+
+import (
+	"sync"
+)
+
+type trieNode struct {
+	c        byte
+	next     [256]*trieNode
+	terminal bool
+}
+
+type KV[V any] struct {
+	data Geche[string, V]
```

Comment:
I don't understand, why it should improve? trieNode represents a single character in the key.

Reply:
If we store the value inside `trieNode` we can return the value without getting querying another `Geche`.

Reply:
Oh that's what you mean. Yep we can do this too, it will be another type of cache then and not a wrapper.
The problem here is that if we do it as new cache type, we can't wrap it to add some new features (e.g. with NewUpdater). Because geche interface does not have `ListByPrefix`. But with a wrapper we can combine e.g `KV(Updater(TTL))`.